pkg/build/controller/common/buildpodutil_test.go

```diff
@@ -1,21 +1,11 @@
 package common
 
-import (
-	"reflect"
-	"testing"
-
-	"k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/runtime/schema"
-	"k8s.io/apimachinery/pkg/util/diff"
-
-	u "github.com/openshift/origin/pkg/build/controller/common/testutil"
-
-	_ "github.com/openshift/origin/pkg/build/apis/build/install"
-)
+// TODO: For some reason the comparison here is missing Kind and ApiVersion fields
```

Comment:
@deads2k @bparees have no idea why...

Reply:
you mean the DeepEquals is failing because the kind/version fields are empty on one of the objects?

Reply:
is build.AsBuild() returning an internal object?

Reply:
it is not internal, still looking

Reply:
(the output looks good visually, so I won't block this PR on this)

Reply:
Make a bugzilla to restore it.

Reply:
Isn't that b/c previously we were working with internals, which don't have these fields set and now that we're working with externals they are always there?

Reply:
This is similar in create printing, where we explicitly set these values, otherwise printing fails.

Reply:
even if you can't maintain the DeepEqual comparison, i'd like to see this test re-enabled w/ some direct field checks so at least we know that GetBuildFromPod and SetBuildInPod works.

Reply:
@bparees I managed to fix it, just added Kind and APIVersion into the test object...